backends/dpdk/dpdkHelpers.cpp

```diff
@@ -765,6 +765,33 @@ bool ConvertStatementToDpdk::preorder(const IR::MethodCallStatement *s) {
             } else {
                 ::error("%1%: unhandled function", s);
             }
+        } else if (a->method->name == "mirror_packet") {
+            auto slot_id = a->expr->arguments->at(0)->expression;
+            auto session_id = a->expr->arguments->at(1)->expression;
+            const IR::Expression *slot_idexpr = slot_id;
+            const IR::Expression *session_idexpr = session_id;
+            // mirror instruction supports only metadata as operands, move any constant arguments
+            // to metadata fields
+            if (slot_id->is<IR::Constant>()){
+                IR::ID tmpName(refmap->newName("mirrorSlot"));
+                BUG_CHECK(metadataStruct, "Metadata structure missing unexpectedly!");
+                metadataStruct->fields.push_back(new IR::StructField(tmpName, slot_id->type));
+                auto tmpMember = new IR::Member(new IR::PathExpression("m"), tmpName);
+                add_instr(new IR::DpdkMovStatement(tmpMember, slot_id));
+                slot_idexpr = tmpMember;
+            }
```

Comment:
are you sure that anything that is not a constant will work?

Reply:
Frontend rejects the program if the arguments to mirror_packet are not compile-time constants. Updated the code to assert on this condition.